collectd_plugin/collectd_plugin.py

```diff
@@ -956,7 +962,5 @@ def _should_filter(values, data, ds):
     data = {}
     collectd.register_init(init)
     collectd.register_config(config, data)
-    if data.get("monitor_endpoint") or (
-            data.get("carbon_host") and data.get("carbon_port")):
```

Comment:
in the config function it seems like we could have a valid writer thread/queue even if we've *only* configured carbon_host/port and not the monitor_endpoint (I don't know if that was a legacy configuration?)

Reply:
>I got the sense maybe controller config missing monitor_endpoint broke the controller metrics - but we fixed that... 

We do have this issue on nodes, if you check the the monitor page for the node. All are empty once we upgrade to new collectd-plugin.

> carbon_host/port and not the monitor_endpoint 

No, I *never* see `carbon_host/port` in collectd configuration. 

I tried to log `_log("endpoint: %s", data.get("monitor_endpoint"))` or `_log("data: %s", data)` in here and  got nothing.

Reply:
>I can't find anything in the collectd python plugin docs that suggest we HAVE to register_write/read ???
> https://collectd.org/documentation/manpages/collectd-python.5.shtml
> https://github.com/signalfx/collectd-example/blob/master/example_plugin.py

I got the samething, and not very clear why I have `monitor_endpoint` in plug-in section, but not able to get these variable in main (outsidf of collectd.regist_XXX fucntions).